### PR TITLE
Add support for selecting resolutions when downloading playlists

### LINKED ISSUE #86: Add an option to choose the resolution when downloading a playlist

### PR DESCRIPTION
**Pull request by brogowski**
Resolves #86

### CODE REVIEW

File: YoutubeDownloader/ViewModels/Dialogs/DownloadMultipleSetupViewModel.cs [hunk 1]
```diff
@@ -12,6 +12,24 @@ namespace YoutubeDownloader.ViewModels.Dialogs
 {
     public class DownloadMultipleSetupViewModel : DialogScreen<IReadOnlyList<DownloadViewModel>>
     {
+        private static readonly string[] AvaliableQualities = new[]
```
Comment (brogowski) [inline YoutubeDownloader/ViewModels/Dialogs/DownloadMultipleSetupViewModel.cs]: I'm not sure how to handle different framerate and resolution combinations... are there any listings for it?

**Comment by Tyrrrz** [comment]
Hi!
What happens now if the user selects a quality that one or more of the videos don't have?
Also, can you please post a screenshot of the download dialog?

**Comment by brogowski** [comment]
> Hi!
> What happens now if the user selects a quality that one or more of the videos don't have?

Original behavior is preserved - best quality is selected. 

> Also, can you please post a screenshot of the download dialog?

image

**Comment by Tyrrrz** [comment]
> Original behavior is preserved - best quality is selected.

I'm thinking it could be a bit misleading if the user selects 2160p but in the end only gets 720p for example. Likewise, if they select 720p and get 1080p instead, it might be undesirable because they could be selecting a lower video quality get smaller output.

Maybe instead add some generic labels?

- Maximum (highest available; 2160p or lower)
- High (up to 1080p)
- Medium (up to 720p)
- Low (up to 480p)
- Minimum (lowest available; 144p or higher)

This way we can match against the majority of changes (categories might need tweaking) without assuming that a specific resolution is going to be available.

**Comment by brogowski** [comment]
> > Original behavior is preserved - best quality is selected.
> 
> I'm thinking it could be a bit misleading if the user selects 2160p but in the end only gets 720p for example. Likewise, if they select 720p and get 1080p instead, it might be undesirable because they could be selecting a lower video quality get smaller output.
> 
> Maybe instead add some generic labels?
> 
> * Maximum (highest available; 2160p or lower)
> * High (up to 1080p)
> * Medium (up to 720p)
> * Low (up to 480p)
> * Minimum (lowest available; 144p or higher)
> 
> This way we can match against the majority of changes (categories might need tweaking) without assuming that a specific resolution is going to be available.

Sure, that's a good idea 👍

**Comment by brogowski** [comment]
image

**Comment by Tyrrrz** [comment]
Thanks!

**Pull request merged**